src/flint/test/test.py

```diff
@@ -1584,6 +1584,212 @@ def test_pickling():
+    test_inplace += fmpz(456)
+    assert test_inplace == F163(123 + 456)
+
+    # Subtraction
+
+    assert F163(123) - F163(456) == F163(123 - 456)
+    assert F163(123) - 456 == F163(123 - 456)
+    assert F163(123) - 456 == F163(123) - F163(456)
+    assert F163(456) - 123 == F163(456) - F163(123)
+    assert 123 - F163(456) == F163(123) - F163(456)
+    assert 456 - F163(123) == F163(456) - F163(123)
+    assert F163(123) - fmpz(456) == F163(123) - F163(456)
+    assert raises(lambda: F163(123) - Fbig(456), ValueError)
+    assert raises(lambda: F163(123) - "AAA", TypeError)
+
+
+    test_inplace = F163(123) 
+    test_inplace -= F163(456) 
+    assert test_inplace == F163(123 - 456)
+
+    test_inplace = F163(123) 
+    test_inplace -= 456
+    assert test_inplace == F163(123 - 456)
+
+    test_inplace = F163(123) 
+    test_inplace -= fmpz(456)
+    assert test_inplace == F163(123 - 456)
+
+    # Multiplication
+
+    assert F163(123) * F163(456) == (123 * 456) % 163
+    assert F163(123) * 456 == (123 * 456) % 163
+    assert 456 * F163(123) == (123 * 456) % 163
+
+    assert F163(1) * F163(123) == F163(1 * 123)
+    assert F163(2) * F163(123) == F163(2 * 123)
+    assert F163(3) * F163(123) == F163(3 * 123)
+    assert 1 * F163(123) == F163(1 * 123)
+    assert 2 * F163(123) == F163(2 * 123)
+    assert 3 * F163(123) == F163(3 * 123)
+    assert F163(123) * 1 == F163(1 * 123)
+    assert F163(123) * 2 == F163(2 * 123)
+    assert F163(123) * 3 == F163(3 * 123)
+    assert fmpz(1) * F163(123) == F163(1 * 123)
+    assert fmpz(2) * F163(123) == F163(2 * 123)
+    assert fmpz(3) * F163(123) == F163(3 * 123)
+    assert raises(lambda: F163(123) * "AAA", TypeError)
+
+
+    test_inplace = F163(123) 
+    test_inplace *= F163(456) 
+    assert test_inplace == F163(123 * 456)
+
+    test_inplace = F163(123) 
+    test_inplace *= 456
+    assert test_inplace == F163(123 * 456)
+
+    test_inplace = F163(123) 
+    test_inplace *= fmpz(456)
+    assert test_inplace == F163(123 * 456)
+
+    # Exponentiation 
+
+    assert F163(0)**0 == pow(0, 0, 163)
+    assert F163(0)**1 == pow(0, 1, 163)
+    assert F163(0)**2 == pow(0, 2, 163)
+    assert raises(lambda: F163(0)**(-1), ZeroDivisionError)
+    assert raises(lambda: F163(0)**("AA"), NotImplementedError)
+
+
+    assert F163(123)**0 == pow(123, 0, 163)
+    assert F163(123)**1 == pow(123, 1, 163)
+    assert F163(123)**2 == pow(123, 2, 163)
+    assert F163(123)**3 == pow(123, 3, 163)
+    assert F163(123)**100 == pow(123, 100, 163)
+    assert F163(123)**(-1) == pow(123, -1, 163)
+    assert F163(123)**(-2) == pow(123, -2, 163)
+    assert F163(123)**(-3) == pow(123, -3, 163)
+    assert F163(123)**(-4) == pow(123, -4, 163)
+
+    # Inversion
+
+    assert raises(lambda: F163(0).__invert__(), ZeroDivisionError)
+    assert F163(123).__invert__() == pow(123, -1, 163)
+    assert F163(1).__invert__() == pow(1, -1, 163)
+    assert F163(2).__invert__() == pow(2, -1, 163)
```

Comment:
This would be like `assert ~F163(123) == pow(123, -1, 163)`.

I think that this use of `~` is confusing but that is what the `__invert__` method is for: we should test the operator rather than the method.